bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -0,0 +1,246 @@
+//! Implements a simple multi-input PSBT signing example
+//!
+//! The purpose of this section is to construct a PSBT that
+//! spends multiple inputs and signs it.
+//! We'll cover the following
+//! [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki)
+//! roles:
+//!
+//! - **Creator**: Creates a PSBT with multiple inputs and outputs.
+//! - **Updater**: Adds Witness and SegWit V0 data to the PSBT.
+//! - **Signer**: Signs the PSBT.
+//! - **Finalizer**: Finalizes the PSBT.
+//!
+//! The example will focus on spending two SegWit V0 inputs:
+//!
+//! 1. 20,000,000 satoshi UTXO, the first receiving ("external") address.
+//! 1. 10,000,000 satoshi UTXO, the first change ("internal") address.
+//!
+//! We'll be sending this to two outputs:
+//!
+//! 1. 25,000,000 satoshis to a receivers' address.
+//! 1. 4,990,000 satoshis back to us as change.
+//!
+//! The miner's fee will be 10,000 satoshis.
+use std::collections::BTreeMap;
+use std::str::FromStr;
+
+use bitcoin::bip32::{ChildNumber, DerivationPath, Fingerprint, Xpriv, Xpub};
+use bitcoin::hashes::Hash;
+use bitcoin::locktime::absolute;
+use bitcoin::psbt::{Input, PsbtSighashType};
+use bitcoin::secp256k1::{Secp256k1, Signing};
+use bitcoin::{
+    consensus, transaction, Address, Amount, Network, OutPoint, Psbt, ScriptBuf, Sequence,
+    Transaction, TxIn, TxOut, Txid, WPubkeyHash, Witness,
+};
+
+// The master xpriv, from which we derive the keys we control.
+const XPRIV: &str = "xprv9tuogRdb5YTgcL3P8Waj7REqDuQx4sXcodQaWTtEVFEp6yRKh1CjrWfXChnhgHeLDuXxo2auDZegMiVMGGxwxcrb2PmiGyCngLxvLeGsZRq";
+
+// The derivation path for the keys we control.
+// This follows the BIP 84 derivation path for Bitcoin.
+const BIP84_DERIVATION_PATH: &str = "m/84'/0'/0'";
+
+// The master fingerprint of the master xpriv.
+const MASTER_FINGERPRINT: &str = "9680603f";
+
+// The dummy UTXO amounts we are spending.
+const DUMMY_UTXO_AMOUNT_INPUT_1: Amount = Amount::from_sat(20_000_000);
+const DUMMY_UTXO_AMOUNT_INPUT_2: Amount = Amount::from_sat(10_000_000);
+
+// The amounts we are sending to someone, and receiving back as change.
+const SPEND_AMOUNT: Amount = Amount::from_sat(25_000_000);
+const CHANGE_AMOUNT: Amount = Amount::from_sat(4_990_000); // 10_000 sat fee.
+
+// Derive the external address xpriv.
+fn get_external_address_xpriv<C: Signing>(
+    secp: &Secp256k1<C>,
+    master_xpriv: Xpriv,
+    index: u32,
+) -> Xpriv {
+    let derivation_path =
+        DerivationPath::from_str(BIP84_DERIVATION_PATH).expect("valid derivation path");
+    let child_xpriv = master_xpriv.derive_priv(secp, &derivation_path).expect("valid child xpriv");
+    let external_index = ChildNumber::from_normal_idx(0).unwrap();
```

Comment:
Also unrelated to this PR but maybe we should have `ZERO_NORMAL` and `ZERO_HARDENED` constants (and variants for `ONE`) because 0 and 1 are super common for hardcoded values. (e.g. see BIP 48, 49, 84 and others... all of which are on my shitlist, but they are super popular and widely deployed.) Would save a lot of unwraps.

Reply:
#2750